SETI/rms-opus
# 417 - Add edit and reorder metadata within galleryView modal

- Fixes #417
- Were any Django, import pipeline, table_schema, or dictionary files modified? Y/N
  - Database used: opus3_test_200517
  - All Django tests pass: Y/N/NA
- Were any JavaScript or CSS files modified? Y
  - JSHINT run on all affected files: Y
  - Tested on Chrome, Firefox, Safari, and iOS: Y
    (Remember to test all browser sizes)
  - Tested for race conditions (with delayed API calls if applicable): Y/N/NA

Description of changes:
-add an 'edit' button on the galleryView modal to allow user to:
   - add a metadata field from the menu list;
   - remove a metadata field;
   - reorder the metadata fields
-Fixed race condition between search widget and selectMetadata modal that cause unselected fields to be highlighted
-Fixed missing scrollbar <bug> in modal

Known problems:
-There is some code that is unused in this pull that will eventually be part of the table header draggable code.  I did not remove it for this pull.
-There are still 3 django tests failing that appear to not be related to these changes but are still under investigation.
